### ISSUE TITLE
GearsBuilder with KeysReader allows to register the function with eventType 'get' BUT doesn't execute the function

### ISSUE THREAD

**Issue by rreddy13Official** [RedisGears/RedisGears#460]
Hi,

I am trying to implement the read-through cache with RedisGears by registering the function with GearBuilder and eventType=get. 
My expectation is that upon 'get' command request, the function executes and fetches the data from thirdparty api(through REST) IF data not already exists in Redis and returns to the request. 

Before implementing I tested with below simple function which will set some test data upon function execution:
```

def add_custom_key(x):
  execute('SET', 'test1', 'value1')  # set key to new value

gb = GearsBuilder('KeysReader')
gb.foreach(add_custom_key)
gb.register(mode='sync', prefix='*', eventTypes=['get'])
```

Adn when
``
gears-cli run --host 127.0.0.1 --port 6379 read-through.py --requirements requirements.txt
``

Got the OK response but function doesn't execute.

If it works only for write commands, I think response should not be OK and can fail with a informative message.

Also documentation says eventType can be any redis command so it would be appreciated if corrected:
![image](https://user-images.githubusercontent.com/75749146/102482275-3f210280-4063-11eb-8d26-efcba4d9f845.png)

**Comment by rreddy13Official** [comment]
We are building application which uses redis cache with Server assisted client side caching() and client library Letttuce(https://lettuce.io/). So any pointers towards achieving Read-through cache using Redis-Gears without losing the feature ClientSideCaching will be helpful.

**Comment by MeirShpilraien** [comment]
Writing the reply here (although I think we talked on discord). The event is on write commands only so you will not get notification about 'get' command. v1.2.0 introduces a new notification called key miss event that allows you to do exactly that. take a look here at how we implemented it with the JVM: https://github.com/RedisGears/rghibernate/tree/write_through_support
```
Read-Through implementation that mainly register on key-miss notification and go fetch the relevant data on such miss. The client is blocked until the data is retrieve and then unblocked and gets the relevant data.
```

@rreddy13Official  Let me know if I can close the issue.
FYI, v1.2.0 is not yet GA.
